integrationTests/vm/arwen/arwenvm/arwenVM_test.go

```diff
@@ -728,3 +731,204 @@ func TestAndCatchTrieError(t *testing.T) {
+	// Scenario:
+	// 1. Deploy FunderSC on shard 0, owned by funderOwner
+	// 2. Deploy ParentSC on shard 1, owned by parentOwner; deployment needs address of FunderSC
+	// 3. parentOwner sends tx to ParentSC with method call 'register' and 42 EGLD (in-shard call, shard 1)
+	// 4. ParentSC emits a cross-shard asyncCall to FunderSC with method 'acceptFunds' and 42 EGLD
+	// 5. assert FunderSC has 42 EGLD
+	zero := big.NewInt(0)
+	transferEGLD := big.NewInt(42)
+
+	testContextFunderSC, err := vm.CreatePreparedTxProcessorWithVMsMultiShard(0, config.EnableEpochs{})
+	require.Nil(t, err)
+	defer testContextFunderSC.Close()
+
+	testContextParentSC, err := vm.CreatePreparedTxProcessorWithVMsMultiShard(1, config.EnableEpochs{})
+	require.Nil(t, err)
+	defer testContextParentSC.Close()
+
+	funderOwner := []byte("12345678901234567890123456789010")
+	require.Equal(t, uint32(0), testContextFunderSC.ShardCoordinator.ComputeId(funderOwner))
+
+	parentOwner := []byte("12345678901234567890123456789011")
+	require.Equal(t, uint32(1), testContextParentSC.ShardCoordinator.ComputeId(parentOwner))
+
+	egldBalance := big.NewInt(1000000000000)
+
+	_, _ = vm.CreateAccount(testContextFunderSC.Accounts, funderOwner, 0, egldBalance)
+	_, _ = vm.CreateAccount(testContextParentSC.Accounts, parentOwner, 0, egldBalance)
+
+	gasPrice := uint64(10)
+	deployGasLimit := uint64(5000000)
+
+	codePath := "../testdata/community"
+	funderCode := codePath + "/funder.wasm"
+	parentCode := codePath + "/parent.wasm"
+
+	logger.ToggleLoggerName(true)
+	// logger.SetLogLevel("*:TRACE")
+
+	// Deploy Funder SC in shard 0
+	funderOwnerAccount, _ := testContextFunderSC.Accounts.LoadAccount(funderOwner)
+	funderAddress := utils.DoDeploySecond(t,
+		testContextFunderSC,
+		funderCode,
+		funderOwnerAccount,
+		gasPrice,
+		deployGasLimit,
+		nil,
+		zero)
+
+	// Deploy Parent SC in shard 1
+	parentOwnerAccount, _ := testContextParentSC.Accounts.LoadAccount(parentOwner)
+	args := [][]byte{[]byte(hex.EncodeToString(funderAddress))}
+	parentAddress := utils.DoDeploySecond(t,
+		testContextParentSC,
+		parentCode,
+		parentOwnerAccount,
+		gasPrice,
+		deployGasLimit,
+		args,
+		zero)
+
+	utils.CleanAccumulatedIntermediateTransactions(t, testContextFunderSC)
+	utils.CleanAccumulatedIntermediateTransactions(t, testContextParentSC)
+
+	// Prepare tx from ParentSC owner to ParentSC (same shard, 1)
+	gasLimit := uint64(5000000)
+	tx := vm.CreateTransaction(1,
+		transferEGLD,
+		parentOwner,
+		parentAddress,
+		gasPrice,
+		gasLimit,
+		[]byte("register"))
+
+	// execute on the sender shard, which emits an async call
+	// from ParentSC (shard 1) to FunderSC (shard 0)
+	retCode, err := testContextParentSC.TxProcessor.ProcessTransaction(tx)
+	require.Equal(t, vmcommon.Ok, retCode)
+	require.Nil(t, err)
+
+	intermediateTxs := testContextParentSC.GetIntermediateTransactions(t)
+	require.NotEmpty(t, intermediateTxs)
+
+	_, err = testContextParentSC.Accounts.Commit()
```

Comment:
from linter: ineffectual assignment to `err` (ineffassign) here
and on next line